### PR TITLE
[profile-id] E2E tests

### PR DESCRIPTION
**Pull request by vguerci**

### CODE REVIEW

File: CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkCaptor.m [hunk 1]
```diff
@@ -33,13 +33,13 @@ @interface CR_NetworkCaptor ()
...
 
 - (instancetype)initWithNetworkManager:(CR_NetworkManager *)networkManager {
-  MockWKWebView *webView = [[MockWKWebView alloc] init];
-  CR_ThreadManager *threadManager = [[CR_ThreadManager alloc] init];
-  CR_DeviceInfo *deviceInfo = [[CR_DeviceInfo alloc] initWithThreadManager:threadManager
-                                                               testWebView:webView];
-  NSURLSession *session = [NSURLSession
-      sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
-  self = [super initWithDeviceInfo:deviceInfo session:session threadManager:threadManager];
+  // FIXME This is a decorator and network manager should be a protocol. This would allow to not
```
Comment (Amokrane) [inline CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkCaptor.m:36]: When is the FIXME planned to be worked on? JIRA?

File: CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkManagerSimulator.m [hunk 2]
```diff
@@ -54,13 +54,14 @@ + (NSTimeInterval)interstitialTtl {
...
 
 - (instancetype)initWithConfig:(CR_Config *)config {
-  MockWKWebView *webView = [[MockWKWebView alloc] init];
-  CR_ThreadManager *threadManager = [[CR_ThreadManager alloc] init];
-  CR_DeviceInfo *deviceInfo = [[CR_DeviceInfo alloc] initWithThreadManager:threadManager
-                                                               testWebView:webView];
-  NSURLSession *session = [NSURLSession
-      sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
-  if (self = [super initWithDeviceInfo:deviceInfo session:session threadManager:threadManager]) {
+  // FIXME This is a different implementation and network manager should be a protocol. This would
```
Comment (Amokrane) [inline CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkManagerSimulator.m:57]: When is the FIXME planned to be worked on? JIRA?

File: CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkCaptor.m (hunk 1 above)
Comment (TurpIF) [reply CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkCaptor.m]: Added JIRA ticket

File: CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkManagerSimulator.m (hunk 2 above)
Comment (TurpIF) [reply CriteoPublisherSdk/Tests/Utility/NetworkManager/CR_NetworkManagerSimulator.m]: Added JIRA ticket

**Comment by TurpIF** [comment]
> Are the commits going to be squashed before merging? Or should the squash be done manually? (I wouldn't advise merging these commits as they are).

I think we may squash this PR. But before this, I need to understand why after updating a comment, 3 integration tests are always failing.

**Comment by TurpIF** [comment]
Three tests are failing making the squash&merge impossible:
- `CR_StandaloneBannerFunctionalTests#test_givenBannerWithGoodAdUnitId_whenLoadAd_thenAdIsLoadedProperly`
- `CR_MopubBannerFunctionalTests#test_givenValidBanner_whenLoadingMopubBanner_thenMopubViewContainsCreative`
- `CR_DfpBannerFunctionalTests#test_givenValidBanner_whenLoadingDfpBanner_thenDfpViewContainsCreative`

The three are failing because the tests check that the AJS creative is well loaded inside the webview. The webview contains the `<script>` tag with the display URL, but the webview don't seem to fetch the creative.

@Amokrane: Those tests are failing on the `main` branch. Also this PR is based from the `3.9.0` tag which is failing too.
I really don't know how to fix such tests. We may ignore them and fix them later, or we may wait for @vguerci to come back and see if he has any clue on this.

**Review by TurpIF — DISMISSED** [review]

**Review by TurpIF — APPROVED** [review]

**Pull request merged**